server/src/test/java/org/opensearch/extensions/rest/RestExecuteOnExtensionTests.java

```diff
@@ -0,0 +1,94 @@
+
+import org.opensearch.rest.RestStatus;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.BytesStreamInput;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+
+public class RestExecuteOnExtensionTests extends OpenSearchTestCase {
+
+    public void testRestExecuteOnExtensionRequest() throws Exception {
+        Method expectedMethod = Method.GET;
+        String expectedUri = "/test/uri";
+        RestExecuteOnExtensionRequest request = new RestExecuteOnExtensionRequest(expectedMethod, expectedUri);
+
+        assertEquals(expectedMethod, request.getMethod());
+        assertEquals(expectedUri, request.getUri());
+
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            request.writeTo(out);
+            out.flush();
+            try (BytesStreamInput in = new BytesStreamInput(BytesReference.toBytes(out.bytes()))) {
+                request = new RestExecuteOnExtensionRequest(in);
+
+                assertEquals(expectedMethod, request.getMethod());
+                assertEquals(expectedUri, request.getUri());
+            }
+        }
+    }
+
+    public void testRestExecuteOnExtensionResponse() throws Exception {
+        RestStatus expectedStatus = RestStatus.OK;
+        String expectedContentType = BytesRestResponse.TEXT_CONTENT_TYPE;
+        String expectedResponse = "Test response";
+        byte[] expectedResponseBytes = expectedResponse.getBytes(StandardCharsets.UTF_8);
+
+        RestExecuteOnExtensionResponse response = new RestExecuteOnExtensionResponse(expectedStatus, expectedResponse);
+
+        assertEquals(expectedStatus, response.getStatus());
+        assertEquals(expectedContentType, response.getContentType());
+        assertArrayEquals(expectedResponseBytes, response.getContent());
+        assertEquals(0, response.getHeaders().size());
+
+        String headerKey = "foo";
+        List<String> headerValueList = List.of("bar", "baz");
+        Map<String, List<String>> expectedHeaders = Map.of(headerKey, headerValueList);
+
+        response = new RestExecuteOnExtensionResponse(expectedStatus, expectedContentType, expectedResponseBytes, expectedHeaders);
+
+        assertEquals(expectedStatus, response.getStatus());
+        assertEquals(expectedContentType, response.getContentType());
+        assertArrayEquals(expectedResponseBytes, response.getContent());
+
+        assertEquals(1, expectedHeaders.keySet().size());
+        assertTrue(expectedHeaders.containsKey(headerKey));
+
+        List<String> fooList = expectedHeaders.get(headerKey);
+        assertEquals(2, fooList.size());
+        assertTrue(fooList.containsAll(headerValueList));
+
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
```

Comment:
This is great. Now we are testing input/output byte stream as well. Can we create an issue to test the other request/response as well we have for extensibility?

Reply:
It's a great thing that we're testing it, as when my test initially failed I discovered `write(byte[])` is different than `writeByteArray(byte[])`.  :-)

Yes, we should test all the other request/response code similarly.  There's a small list of other fixes I'd like (compiler warnings, etc.) that I'll put in an issue later this week.